lib/main.dart

```diff
@@ -9,82 +9,94 @@ import 'package:quick_log/quick_log.dart';
 import 'package:sentry/sentry.dart';
 import 'package:uuid/uuid.dart';
 import 'package:http/http.dart' as http;
+import 'package:args/args.dart';
 
 import 'controllers/main_info.dart';
 
-String pathToSketchFile;
 String resultsDirectory = Platform.environment['SENTRY_DSN'] ?? '/temp';
 final SentryClient sentry = SentryClient(dsn: resultsDirectory);
+ArgResults argResults;
 
 void main(List<String> args) async {
   await checkConfigFile();
 
+  //Sentry logging initialization
   MainInfo().sentry = SentryClient(dsn: resultsDirectory);
   var log = Logger('Main');
-
   log.info(args.toString());
 
   MainInfo().cwd = Directory.current;
 
-  var path = '';
-  var projectName = '';
-  var designType = 'sketch';
-  final _helpText = "Usage Options: \n" +
-      "-p \t Path to the sketch File \n" +
-      "-o \t Output Path\n" +
-      "-n \t Name of the project\n" +
-      "-c \t Path of the configuration file\n";
-
-  //Check if no args passed or only -h passed
-  //If arguments is empty or only has -h
-  if (args.length == 0 || args[0] == '-h') {
-    print(_helpText);
-    return;
+  ///sets up parser
+  final parser = ArgParser()
+    ..addOption('path',
+        help: 'Path to the design file', valueHelp: 'path', abbr: 'p')
+    ..addOption('out', help: 'The output path', valueHelp: 'path', abbr: 'o')
+    ..addOption('project-name',
+        help: 'The name of the project', abbr: 'n', defaultsTo: 'temp')
+    ..addOption('config-path',
+        help: 'Path of the configuration file',
+        abbr: 'c',
+        defaultsTo: 'lib/configurations/configurations.json')
```

Comment:
For the default configuration, this should be `default:lib/configurations/configurations.json` 😄